filters/cropbyfocalarea.go

```diff
@@ -0,0 +1,114 @@
+package filters
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+	"github.com/danpersa/bimg"
+	"strconv"
+)
+
+// CropByFocalAreaName is the name of the filter
+const CropByFocalAreaName = "cropByFocalArea"
+
+type cropByFocalArea struct {
+}
+
+// NewCropByFocalArea creates a new filter of this type
+func NewCropByFocalArea() filters.Spec {
+	return &cropByFocalArea{}
+}
+
+func (f *cropByFocalArea) Name() string {
+	return CropByFocalAreaName
+}
+
+func (f *cropByFocalArea) CreateOptions(imageContext *ImageFilterContext) (*bimg.Options, error) {
+	log.Debug("Create options for crop by focal area ", f)
+
+	imageSize, err := imageContext.Image.Size()
+
+	if err != nil {
+		return nil, err
+	}
+
+	desired_width, err := strconv.Atoi(imageContext.PathParam("desired_width"))
+
+	if err != nil {
+		return nil, err
+	}
+
+	desired_height, err := strconv.Atoi(imageContext.PathParam("desired_height"))
+
+	if err != nil {
+		return nil, err
+	}
+
+	focalPointX, err := strconv.Atoi(imageContext.PathParam("focalPointX"))
+
+	if err != nil {
+		return nil, err
+	}
+
+	focalPointY, err  := strconv.Atoi(imageContext.PathParam("focalPointY"))
+
+	if err != nil {
+		return nil, err
+	}
+
+	if desired_width > imageSize.Width {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	if desired_height > imageSize.Height {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	if float64(focalPointX) + 0.5*float64(desired_width) > float64(imageSize.Width) {
+		focalPointX = imageSize.Width - int(0.5*float64(desired_width))
+	} else if float64(focalPointX) - 0.5*float64(desired_width) < 0 {
+		focalPointX = desired_width/2
+	}
+
+	if float64(focalPointY) + 0.5*float64(desired_height) > float64(imageSize.Height) {
+		focalPointY = imageSize.Height - int(0.5*float64(desired_height))
+	} else if float64(focalPointY) - 0.5*float64(desired_height) < 0 {
+		focalPointY = desired_height/2
+	}
+
+	return &bimg.Options{
+		AreaWidth:  desired_width,
+		AreaHeight: desired_height,
+		Top:    focalPointY - desired_height/2,
+		Left:    focalPointX - desired_width/2,
+		}, nil
+}
+
+func (f *cropByFocalArea) CanBeMerged(other *bimg.Options, self *bimg.Options) bool {
+	return (other.AreaWidth == 0 || other.AreaWidth == self.AreaWidth) &&
```

Comment:
Why not returning false? Can you explain the logic?

```suggestion
	return false
```

Reply:
The idea was to allow the filter to be merged in any other case that is not dealing with the cropping specifically. But if you think is better to never allow merge, for sure we can go in this direction

Reply:
I would not for not merging at all, as applying cropping two times is error prone and should probably never happen.